### ISSUE TITLE
Toaster (static usage): Toast Notifications no longer appear at bottom of a scrolled page (regression)

### ISSUE THREAD

**Issue by joelbohnes** [palantir/blueprint#4864]
#### Environment

- __Package version(s)__: 3.48.0
- __Operating System__:  Windows 10
- __Browser name and version__: Google Chrome 92.0.4515.159

#### Code Sandbox
https://codesandbox.io/s/vibrant-waterfall-9bjnr

#### Steps to reproduce
1. Use toaster component statically (toaster = Toaster.create(...)) with positioning set to bottom
2. Display a toast (toaster.show(..)) while the page is scrolled

#### Actual behaviour
The toast notification appears at a location that isn't the bottom of the viewport.
This behaviour seems to have been introduced in version 3.40.0.

#### Expected behaviour
Expected the behaviour of previous blueprintjs versions (i.e. 3.39.0):
- Toast notifications always appear at the bottom of the viewport, regardless of scroll location.
- Position of notification remain at the bottom while scrolling

#### Possible solution
Set toaster container to always use fixed positioning via CSS:
```css
.bp3-toast-container {
  position: fixed !important;
}
```
This may break non-static usage of the toaster component (especially when usePortal=false)

**Comment by adidahiya** [comment]
We released this small breaking change in 3.40.0, see the [changelog](https://github.com/palantir/blueprint/wiki/3.x-Changelog#blueprintjscore-3400):

> - #4555 fix(Toaster): use absolute positioning when inline (usePortal={false})
>   - ⚠️ This is a small CSS breaking change for users of this option, but also clearly a bugfix (the layout would have been broken or required custom positioning before this change).

I don't see this problem happening in the Blueprint docs, so it's likely something you can work around with some CSS. I poked a bit at the sandbox and didn't immediately find a solution, so I'll leave this open for further investigation / need to revisit so that we can provide clear guidance to work around the issue.

**Comment by trfv** [comment]
https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/toast/toaster.tsx#L83-L84
https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/toast/toaster.tsx#L132

`usePortal` is always false when you use `Toaster.create()`. The workaround is stop using `Toaster.create()` I think.

cf) https://blueprintjs.com/docs/#core/components/toast.react-component-usage

**Comment by JoeDuncko** [comment]
Yup, using a react component instead fixed it. I made a cool little `useToasterContext` hook for it:

```Javascript
import { Position, Toaster } from "@blueprintjs/core";
import { createContext, useContext, useRef } from "react";

const ToasterContext = createContext({
    clear: undefined,
    dismiss: undefined,
    getToasts: undefined,
    show: undefined,
});

export function ToasterProvider({ children }) {
    const toasterRef = useRef();

    return (
        <ToasterContext.Provider
            value={{
                clear: () => toasterRef.current?.clear(),
                dismiss: (key) => toasterRef.current?.dismiss(key),
                getToasts: () => toasterRef.current?.getToasts(),
                show: (args, key) => toasterRef.current?.show(args, key),
            }}
        >
            <Toaster position={Position.TOP_RIGHT} ref={toasterRef} />

            {children}
        </ToasterContext.Provider>
    );
}

export const useToasterContext = () => {
    return useContext(ToasterContext);
};
```
